From f50206555992abb802cee4e3f951d1ea669cb8bc Mon Sep 17 00:00:00 2001
From: Sean Wang <sean.wang@mediatek.com>
Date: Mon, 18 Sep 2023 16:03:06 +0800
Subject: [PATCH] wifi: mt76: move struct ieee80211_chanctx_conf up to struct mt76_vif
Git-commit: f50206555992abb802cee4e3f951d1ea669cb8bc
Patch-mainline: v6.7-rc1
References: git-fixes

Move struct ieee80211_chanctx_conf up to struct mt76_vif to allow the
connac2 library can access the struct ieee80211_chanctx_conf * member in
struct mt76_vif.

Signed-off-by: Sean Wang <sean.wang@mediatek.com>
Reviewed-by: David Ruth <druth@chromium.org>
Tested-by: David Ruth <druth@chromium.org>
Signed-off-by: Deren Wu <deren.wu@mediatek.com>
Signed-off-by: Felix Fietkau <nbd@nbd.name>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/net/wireless/mediatek/mt76/mt76.h        |    1 +
 drivers/net/wireless/mediatek/mt76/mt7921/main.c |   12 ++++++------
 2 files changed, 7 insertions(+), 6 deletions(-)

--- a/drivers/net/wireless/mediatek/mt76/mt76.h
+++ b/drivers/net/wireless/mediatek/mt76/mt76.h
@@ -715,6 +715,7 @@ struct mt76_vif {
 	u8 wmm_idx;
 	u8 scan_seq_num;
 	u8 cipher;
+	struct ieee80211_chanctx_conf *ctx;
 };
 
 struct mt76_phy {
--- a/drivers/net/wireless/mediatek/mt76/mt7921/main.c
+++ b/drivers/net/wireless/mediatek/mt76/mt7921/main.c
@@ -804,7 +804,7 @@ void mt7921_mac_sta_assoc(struct mt76_de
 
 	if (vif->type == NL80211_IFTYPE_STATION && !sta->tdls)
 		mt76_connac_mcu_uni_add_bss(&dev->mphy, vif, &mvif->sta.wcid,
-					    true, mvif->ctx);
+					    true, mvif->mt76.ctx);
 
 	ewma_avg_signal_init(&msta->avg_ack_signal);
 
@@ -839,7 +839,7 @@ void mt7921_mac_sta_remove(struct mt76_d
 		if (!sta->tdls)
 			mt76_connac_mcu_uni_add_bss(&dev->mphy, vif,
 						    &mvif->sta.wcid, false,
-						    mvif->ctx);
+						    mvif->mt76.ctx);
 	}
 
 	spin_lock_bh(&dev->sta_poll_lock);
@@ -1628,7 +1628,7 @@ mt7921_start_ap(struct ieee80211_hw *hw,
 	mt7921_mutex_acquire(dev);
 
 	err = mt76_connac_mcu_uni_add_bss(phy->mt76, vif, &mvif->sta.wcid,
-					  true, mvif->ctx);
+					  true, mvif->mt76.ctx);
 	if (err)
 		goto out;
 
@@ -1660,7 +1660,7 @@ mt7921_stop_ap(struct ieee80211_hw *hw,
 		goto out;
 
 	mt76_connac_mcu_uni_add_bss(phy->mt76, vif, &mvif->sta.wcid, false,
-				    mvif->ctx);
+				    mvif->mt76.ctx);
 
 out:
 	mt7921_mutex_release(dev);
@@ -1685,7 +1685,7 @@ static void mt7921_ctx_iter(void *priv,
 	struct mt7921_vif *mvif = (struct mt7921_vif *)vif->drv_priv;
 	struct ieee80211_chanctx_conf *ctx = priv;
 
-	if (ctx != mvif->ctx)
+	if (ctx != mvif->mt76.ctx)
 		return;
 
 	if (vif->type == NL80211_IFTYPE_MONITOR)
@@ -1748,7 +1748,7 @@ static void mt7921_mgd_prepare_tx(struct
 		       jiffies_to_msecs(HZ);
 
 	mt7921_mutex_acquire(dev);
-	mt7921_set_roc(mvif->phy, mvif, mvif->ctx->def.chan, duration,
+	mt7921_set_roc(mvif->phy, mvif, mvif->mt76.ctx->def.chan, duration,
 		       MT7921_ROC_REQ_JOIN);
 	mt7921_mutex_release(dev);
 }
